_source/logzio_collections/_prometheus-sources/dotnet-custom.md

```diff
@@ -0,0 +1,555 @@
+---
+title: Ship custom metrics from your .NET Core application
+logo:
+  logofile: dotnet.svg
+  orientation: horizontal
+data-source: .NET Core application
+contributors:
+  - nshishkin
+shipping-tags:
+  - popular
+order: 280
+---
+
+<!-- tabContainer:start -->
+<div class="branching-container">
+
+* [Overview](#overview)
+* [Hardcoded Logz.io exporter](#in-code)
+* [Logz.io exporter in config](#in-config)
+* [Advanced settings](#advanced-settings)
+{:.branching-tabs}
+
+<!-- tab:start -->
+<div id="overview">
+
+Logzio.App.Metrics is an open-source and cross-platform .NET library used to record metrics within an application and forward it to Logz.io.
+
+These instructions show you how to:
+
+* Create a basic custom metrics export configuration with a hardcoded Logz.io exporter
+* Create a basic custom metrics export configuration with a Logz.io exporter defined by a configuration file
+* Add advanced settings to the basic custom metrics export configuration
+
+
+</div>
+<!-- tab:end -->
+ 
+
+<!-- tab:start -->
+<div id="in-code">
+
+
+#### Send custom metrics to Logz.io with a hardcoded Logz.io exporter
+
+**Before you begin, you'll need**: 
+
+* an application in .NET Core 3.1 and higher
+* an active account with Logz.io
+
+
+<div class="tasklist">
+
+
+##### Install the App.Metrics.Logzio package
+
+
+Install the App.Metrics.Logzio package from the Package Manager Console:
+
+```shell
+Install-Package Logzio.App.Metrics
+```
+
+Alternatively, If you prefer to install the library manually, download the latest version from NuGet Gallery
+
+
+##### Create MetricsBuilder
+
+To create MetricsBuilder, copy and paste the following code into the function of the code that you need to export metris from:
+
+```java
+var metrics = new MetricsBuilder()
+                .Report.ToLogzioHttp("[[<<LISTENER-HOST>>]]", "[[<<METRICS-SHIPPING-TOKEN>>]]")
+                .Build();
+```
+
+{% include log-shipping/listener-var.html %} 
```

Comment:
Please add to the listener examples the 'https://' or 'http://' and the port. If they use 'https://' the port is ':8053' and if they use 'http://' the port is ':8052'